docs/configuration.md

```diff
@@ -0,0 +1,71 @@
+# Configuring Spectral
+
+## Example config file (spectral.yaml)
+
+```yaml
+rules:
+    - rule-name:
+        given: $..parameters[*]
+        then:
+            field: description
+            function: truthy        
+```
+
+## Rules
+
+Rules are highly configurable. There are only few required parameters but the optional ones gives powerful flexibility.
+
+TBD: replace TS interfaces with a nice table.
+
+```ts
+export interface IRule<T = string, O = any> {
+  type?: RuleType;
```

Comment:
Why is this optional?

Reply:
Out of scope, copy paste from types.ts

I'm happy to discuss this but it's not the main purpose of this PR